rpmio/macro.cc

```diff
@@ -1992,10 +1993,20 @@ rpmExpand(const char *arg, ...)
 int
 rpmExpandNumeric(const char *arg)
 {
+    int res;
     if (arg == NULL)
 	return 0;
     auto [ ign, val ] = macros().expand_numeric(arg);
-    return val;
+    if (val > INT_MAX) {
+	rpmlog(RPMLOG_WARNING, _("Macro value too big for int: %" PRIu64 " Using %i instead.\n"), val, INT_MAX);
+	res = INT_MAX;
+    } else if (val < INT_MIN) {
+	    rpmlog(RPMLOG_WARNING, _("Macro value too small for int: %" PRIu64 " Using %i instead.\n"), val, INT_MIN);
+	res = INT_MIN;
+    } else {
```

Comment:
Turns out there's an STL solution for this: https://en.cppreference.com/w/cpp/algorithm/clamp

No point having separate messages for too small and too big, just state that it's outside range.